### DISCUSSION TITLE
Lupus LS12128 roller shutter

### DISCUSSION THREAD

**Discussion by cody82** [Koenkk/zigbee2mqtt#6529] [Devices]
I got a LS12128 roller shutter which basically works fine with zigbee2mqtt. I can OPEN/CLOSE the shutters. But the position command is wierd. It seems pretty random.
The Lupus manual tells me I have to configure I have to configure the duration for OPEN/CLOSE in the gateway. Of course I dont have one. How should I do this with zigbee2mqtt?
Also there is no STOP command?

**Comment by kds69** [comment]
Hi,
Lucky you...I just got same roller shutter relay and inspired by Climax SCMZB technical instruction, I just gave a try.
In a nutshell:
1-Open fully the shutter
2-Make sure none of the physical button of corresponding wall switch is left pressed down
2-Press 3-10s the LUPUS relay connection button
3-LED will start to blink
4-At a certain point relay will reconnect to zigbee coordinator (hear a "click") but it will still blink = it expects to learn closing time and opening time
5-Press down button
6-Just when shutter is fully closed then press "up" button (sometimes I have to press twice).
7-Just when shutter is fully opened then press "down" button (or release up button)
8-Et voilà! Relay will now have learnt how long it takes to travel top-bottom and vice-et-versa (hear a "click" telling relay circuit is open) 

From then onwards, you will remark that when you send commands the relay will emit a "click" when it is done with commands, while before it would take 4mn.
You can also try by pressing both up and down button together to initiate a new learning cycle... with very recent firmware (I can't tell more nor I had a chance to test)

**Reply by cody82** [reply]
Thank you, will try that!!

**Comment by renne** [comment]
@kds69 Please add the information about calibrating the up/down runtime to the [device page of the Lupus LS12128](https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/devices/LS12128.md).

**Reply by kds69** [reply]
Please go ahead and do it. I don't know how to do it.

**Reply by renne** [reply]
[Done](https://github.com/Koenkk/zigbee2mqtt.io/pull/797)
